### PR TITLE
Filter story pins from user input

### PR DESCRIPTION
**Pull request by DinuWije**
## Description 
Taking user input to filter pins onto the map.

## Demo
image
image

### CODE REVIEW

File: ui/src/components/Filter.tsx
Comment (AbhiPrasad) [inline ui/src/components/Filter.tsx]: Why do we need to pass `state` into the filter?

Can we just expose `onChange`, and move the `onTagsChange` function up to the parent component? Also please type the props?

so something like:

```
export function Filter({ onChange }: FilterProps): JSX.Element {
```

File: ui/src/pages/ShoeMap.tsx
Comment (AbhiPrasad) [inline ui/src/pages/ShoeMap.tsx]: This is not an array of stories, it's an array of countries, so we should probably rename accordingly.

File: ui/src/pages/ShoeMap.tsx
Comment (AbhiPrasad) [inline ui/src/pages/ShoeMap.tsx]: 1) You can find the intersection between two arrays with `.filter` and `.includes`. So something like:
```
const filteredStoryArr = stories.filter(value => filterStories.includes(value.author_country));
```
^ This is not super optimum, but works well for our case. If you are wondering about performance, we can also use es6 sets (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) or memoize with a hash map.

2) We shouldn't need to call a anon function with `(() => {...})()`. Instead we can compute this before the return value. So something like:

```js
const { data, error } = useSWR<Story[]>("/api/stories");
const [filteredCountries, setFilteredCountries] = useState([]);

...

// If no filtered countries just make stories equal to data
// otherwise filter the stories based on the intersection of the two arrays.
const stories = filteredCountries.length === 0 ? data : data.filter(story => filterStories.includes(story.author_country));

...

return (
  ...
  ...
  
  <PinCluster stories={stories} openDrawer={handleOpenDrawer} />
  ...
  ...
```

**Review by AbhiPrasad — CHANGES_REQUESTED** [review]
Nice! Lmk if you have any questions about my review.

Oh yeah make sure you run `yarn fix` in the ui folder to make CI go green (don't merge on red CI).

File: ui/src/pages/ShoeMap.tsx
Comment (AbhiPrasad) [inline ui/src/pages/ShoeMap.tsx:115]: Can we create a Github issue so we keep track of render stuff in error states?

Will require some design work as well so it we don't gotta do it immediately.

File: ui/src/components/Filter.tsx
Comment (AbhiPrasad) [inline ui/src/components/Filter.tsx]: Let's change the type here. This means that:

```suggestion
          onChange={onChange}
```

```ts
export interface FilterProps {
  onChange: (event: idkWhatEventTypeIs, value: string[]) => void
}
```

Then you'll have to change:

```ts
// ui/src/pages/ShoeMap.tsx 
function onTagsChange(_, value: string[]) {
  setFilteredCountries(options);
}
```

**Review by AbhiPrasad — COMMENTED** [review]
1 more thing, I'll approve after this. Good stuff!

File: ui/src/pages/ShoeMap.tsx
Comment (DinuWije) [reply ui/src/pages/ShoeMap.tsx:115]: Done!

**Review by AbhiPrasad — APPROVED** [review]
🚢 🇮🇹   (make sure to squash and merge)

**Pull request merged**
